test/integration/functional_test.go

```diff
@@ -2308,3 +2309,27 @@ func validateLicenseCmd(ctx context.Context, t *testing.T, _ string) {
 		t.Errorf("expected license file to contain %q, but was not found", expectedString)
 	}
 }
+
+// validateInvalidService makes sure minikube will not start a tunnel for an unavailable service that has no running pods
+func validateInvalidService(ctx context.Context, t *testing.T, profile string) {
+
+	// try to start an invalid service. This service is linked to a pod whose image name is invalid, so this pod will never become running
+	rrApply, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "apply", "-f", filepath.Join(*testdataDir, "invalidsvc.yaml")))
+	defer func() {
+		// Cleanup test configurations in advance of future tests
+		rr, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "delete", "-f", filepath.Join(*testdataDir, "invalidsvc.yaml")))
+		if err != nil {
+			t.Fatalf("clean up %s failed: %v", rr.Command(), err)
+		}
+	}()
+	if err != nil {
+		t.Fatalf("%s failed: %v", rrApply.Command(), err)
+	}
```

Comment:
```suggestion
	rrApply, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "apply", "-f", filepath.Join(*testdataDir, "invalidsvc.yaml")))
	if err != nil {
		t.Fatalf("%s failed: %v", rrApply.Command(), err)
	}
	defer func() {
		// Cleanup test configurations in advance of future tests
		rr, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "delete", "-f", filepath.Join(*testdataDir, "invalidsvc.yaml")))
		if err != nil {
			t.Fatalf("clean up %s failed: %v", rr.Command(), err)
		}
	}()
```

Check for error before defer because if apply failed, delete is not needed